ambry-server/src/integration-test/java/com.github.ambry.server/CloudAndStoreReplicationTest.java

```diff
@@ -0,0 +1,325 @@
+        new MockDataNodeId("localhost", vcrPortList, Collections.singletonList(vcrMountPath), cloudDc);
+
+    // create ambry server recovery cluster
+    MockClusterMap serverClusterMap = new MockClusterMap(false, 2, 1, 1, true);
+    recoveryCluster = new MockCluster(serverClusterMap, Collections.singletonList(vcrNode), recoveryProperties);
+    partitionId = recoveryCluster.getClusterMap().getWritablePartitionIds(null).get(0);
+    allRecoveryNodes = serverClusterMap.getDataNodes();
+
+    // record ambry server node which will get partition leadership notification.
+    partitionLeaderRecoveryNode = allRecoveryNodes.get(0);
+    MockClusterAgentsFactory leaderMockClusterAgentsFactory = new MockClusterAgentsFactory(serverClusterMap,
+        serverClusterMap.getAllPartitionIds(null).stream().map(PartitionId::toPathString).collect(Collectors.toList()));
+
+    // Start Helix Controller and ZK Server.
+    int zkPort = 31999;
+    String zkConnectString = "localhost:" + zkPort;
+    String vcrClusterName = "vcrTestCluster";
+    zkInfo = new TestUtils.ZkInfo(TestUtils.getTempDir("helixVcr"), cloudDc, (byte) 1, zkPort, true);
+    helixControllerManager =
+        VcrTestUtil.populateZkInfoAndStartController(zkConnectString, vcrClusterName, recoveryCluster.getClusterMap());
+
+    Properties vcrProperties =
+        VcrTestUtil.createVcrProperties(vcrNode.getDatacenterName(), "vcrTestCluster", zkConnectString, 12310, 12410,
+            null);
+    vcrProperties.putAll(recoveryProperties);
+    MockNotificationSystem notificationSystem = new MockNotificationSystem(recoveryCluster.getClusterMap());
+
+    // Create blobs and data for upload to vcr.
+    int blobCount = 10;
+    cloudBlobIds =
+        ServerTestUtil.createBlobIds(blobCount, recoveryCluster.getClusterMap(), accountId, containerId, partitionId);
+    serverBlobIds =
+        ServerTestUtil.createBlobIds(blobCount, recoveryCluster.getClusterMap(), accountId, containerId, partitionId);
+
+    // Create cloud destination and start vcr server.
+    latchBasedInMemoryCloudDestination = new LatchBasedInMemoryCloudDestination(cloudBlobIds);
+    CloudDestinationFactory cloudDestinationFactory =
+        new LatchBasedInMemoryCloudDestinationFactory(latchBasedInMemoryCloudDestination);
+    vcrServer =
+        VcrTestUtil.createVcrServer(new VerifiableProperties(vcrProperties), recoveryCluster.getClusterAgentsFactory(),
+            notificationSystem, cloudDestinationFactory);
+    vcrServer.startup();
+
+    // initialize and start ambry servers
+    for (MockDataNodeId serverNode : allRecoveryNodes) {
+      if (serverNode.equals(partitionLeaderRecoveryNode)) {
+        recoveryCluster.initializeServer(serverNode, recoveryProperties, false, true, notificationSystem,
+            SystemTime.getInstance(), leaderMockClusterAgentsFactory);
+      } else {
+        recoveryCluster.initializeServer(serverNode, recoveryProperties, false, true, notificationSystem,
```

Comment:
After consolidating the two initializeServer methods, you can consolidate this into one call passing:
    serverNode.equals(partitionLeaderRecoveryNode) ? leaderMockClusterAgentsFactory : null